src/api/app/models/workflow_run.rb

```diff
@@ -67,6 +67,17 @@ def event_source_url
     payload.dig(*mapped_source_url) if mapped_source_url
   end
 
+  def generic_event_type
+    # We only have filters for push, tag_push, and pull_request
+    if hook_event == 'Push Hook' || payload.fetch('ref', '').match('refs/heads')
```

Comment:
Is that `github  || gitlab`?

Reply:
Or why are we doing this?

Reply:
`Push Hook` represents `gitlab` and `refs/heads` represents `github` for push events. We need this because in the view we don't differentiate b/w github and gitlab events. That's why we made a generic event to represent `push`

Reply:
Okay RTFMd...

Reply:
> Okay and for github there is only `push` no matter if it's a push ref or push tag?

Exactly, for Github it's `push` for both `tag` and `push` events. The only difference is in `ref`